app/models/comment.rb

```diff
@@ -35,17 +38,30 @@ class Comment < ActiveRecord::Base
 
   after_create :call_after_commented
 
-  def self.build(commentable, user, body, p_id=nil)
-    new commentable: commentable,
-        user_id:     user.id,
-        body:        body,
-        parent_id:   p_id
+  after_initialize do |comment| 
```

Comment:
Why did you use an `after_initialize` hook instead of setting 0 as a default value in database? :open_mouth: